morpheus/models/dfencoder/dataloader.py

```diff
@@ -352,89 +349,46 @@ def get_preloaded_data(self):
-        """A helper function to get a train dataset with the provided parameters.
+    @property
+    def preprocess_fn(self):
+        return self._preprocess_fn
 
-        Parameters
-        ----------
-        model : AutoEncoder
-            The autoencoder model used to get relevant params and the preprocessing func.
-        data_folder : str
-            The path to the folder containing the data.
-        load_data_fn : function, optional
-            A function for loading data from a provided file path into a pandas.DataFrame, by default pd.read_csv.
-        preload_data_into_memory : bool, optional
-            Whether to preload all the data into memory, by default False.
+    @preprocess_fn.setter
+    def preprocess_fn(self, value):
+        self._preprocess_fn = value
 
-        Returns
-        -------
-        DatasetFromPath
-            Validation Dataset set up to load from the path.
-        """
-        dataset = DatasetFromPath(
-            data_folder,
-            model.batch_size,
-            model.preprocess_train_data,
-            load_data_fn=load_data_fn,
-            shuffle_rows_in_batch=True,
-            shuffle_batch_indices=True,
-            preload_data_into_memory=preload_data_into_memory,
-        )
-        return dataset
+    @property
+    def batch_size(self):
+        return self._batch_size
 
-    @staticmethod
-    def get_validation_dataset(model, data_folder, load_data_fn=pd.read_csv, preload_data_into_memory=True):
-        """A helper function to get a validation dataset with the provided parameters.
+    @batch_size.setter
+    def batch_size(self, value):
+        self._batch_size = value
 
-        Parameters
-        ----------
-        model : AutoEncoder
-            The autoencoder model used to get relevant params and the preprocessing func.
-        data_folder : str
-            The path to the folder containing the data.
-        load_data_fn : function, optional
-            A function for loading data from a provided file path into a pandas.DataFrame, by default pd.read_csv.
-        preload_data_into_memory : bool, optional
-            Whether to preload all the data into memory, by default True.
-            (can speed up data loading if the data can fit into memory)
+    @property
+    def shuffle_rows_in_batch(self):
+        return self._shuffle_rows_in_batch
 
-        Returns
-        -------
-        DatasetFromPath
-            Validation Dataset set up to load from the path.
-        """
-        dataset = DatasetFromPath(
-            data_folder,
-            model.eval_batch_size,
-            model.preprocess_validation_data,
-            load_data_fn=load_data_fn,
-            shuffle_rows_in_batch=False,
-            preload_data_into_memory=preload_data_into_memory,
-        )
-        return dataset
+    @shuffle_rows_in_batch.setter
+    def shuffle_rows_in_batch(self, value):
+        self._shuffle_rows_in_batch = value
```

Comment:
Again, do we need the setter for this property?

Reply:
We don't need a 'setter' necessarily, but we do need to be able to set it; the use case is when transitioning from training to validation.